tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/PipelineConvention.cs

```diff
@@ -349,23 +349,19 @@ protected bool EnsureDefautPullRequestTrigger(BuildDefinition definition, bool o
                 {
                     if (overrideYaml)
                     {
-                        if (trigger.SettingsSourceType != 1 ||
-                            trigger.BranchFilters.Contains("+*"))
+                        // Override what is in the yaml file and use what is in the pipeline definition
+                        trigger.SettingsSourceType = 1;
+
+                        if (!trigger.BranchFilters.Contains("+*"))
                         {
-                            // Override what is in the yaml file and use what is in the pipeline definition
-                            trigger.SettingsSourceType = 1;
                             trigger.BranchFilters.Add("+*");
```

Comment:
It wasn't here before but do we need a hasChanges=true in this if block?

Reply:
Updated.